controllers/sriovnetworkpoolconfig_controller.go

```diff
@@ -0,0 +1,205 @@
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			reqLogger.Info("instance not found")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// examine DeletionTimestamp to determine if object is under deletion
+	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object. This is equivalent
+		// registering our finalizer.
+		if !sriovnetworkv1.StringInArray(sriovnetworkv1.POOLCONFIGFINALIZERNAME, instance.ObjectMeta.Finalizers) {
+			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, sriovnetworkv1.POOLCONFIGFINALIZERNAME)
+			if err := r.Update(context.Background(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		if utils.ClusterType == utils.ClusterTypeOpenshift {
+			if err = r.syncOvsHardwareOffloadMachineConfigs(instance, false); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		// The object is being deleted
+		if sriovnetworkv1.StringInArray(sriovnetworkv1.POOLCONFIGFINALIZERNAME, instance.ObjectMeta.Finalizers) {
+			// our finalizer is present, so lets handle any external dependency
+			reqLogger.Info("delete SriovNetworkPoolConfig CR", "Namespace", instance.Namespace, "Name", instance.Name)
+			if utils.ClusterType == utils.ClusterTypeOpenshift {
+				if err = r.syncOvsHardwareOffloadMachineConfigs(instance, true); err != nil {
+					// if fail to delete the external dependency here, return with error
+					// so that it can be retried
+					return reconcile.Result{}, err
+				}
+			}
+			// remove our finalizer from the list and update it.
+			instance.ObjectMeta.Finalizers = sriovnetworkv1.RemoveString(sriovnetworkv1.POOLCONFIGFINALIZERNAME, instance.ObjectMeta.Finalizers)
+			if err := r.Update(context.Background(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *SriovNetworkPoolConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&sriovnetworkv1.SriovNetworkPoolConfig{}).
+		Complete(r)
+}
+
+func (r *SriovNetworkPoolConfigReconciler) syncOvsHardwareOffloadMachineConfigs(nc *sriovnetworkv1.SriovNetworkPoolConfig, deletion bool) error {
+	logger := r.Log.WithName("syncOvsHardwareOffloadMachineConfigs")
+
+	data := render.MakeRenderData()
+	mcpMap := make(map[string]bool)
+
+	mcpList := &mcfgv1.MachineConfigPoolList{}
+	err := r.List(context.TODO(), mcpList, &client.ListOptions{})
```

Comment:
Why we fetch the whole list of MCPs, instead of getting the MCP by `Spec.OvsHardwareOffloadConfig.Name`? We have a one-to-one mapping between SriovNetworkPoolConfig, MCP CR here, don't we?

Reply:
Done, changed to get MCP according to Spec.OvsHardwareOffloadConfig.Name.